### DISCUSSION TITLE
Why does IFocusZoneProps extend React.HTMLAttributes<HTMLElement | FocusZone>?

### DISCUSSION THREAD

**Discussion by Sentri** [microsoft/fluentui#16270] [Q&A] [answered]
Hello,

Why does the `IFocusZoneProps` interface extend `React.HTMLAttributes<HTMLElement | FocusZone>` instead of just plain `React.HTMLAttributes<HTMLElement>`? 

https://github.com/microsoft/fluentui/blob/ac24a6a79a7f09559bb35ffbc5b28f53ecd9e95f/packages/react-focus/src/components/FocusZone/FocusZone.types.ts#L46

This makes it so to react to events like `onKeyDown` we have to implement a type guard or some kind of a cast downstream to get to the target `HTMLElement` value. It seems like when the props are called in the code, the argument is always a `React.KeyboardEvent<HTMLElement>`, and never `FocusZone`.

See here:
https://github.com/microsoft/fluentui/blob/ac24a6a79a7f09559bb35ffbc5b28f53ecd9e95f/packages/react-focus/src/components/FocusZone/FocusZone.tsx#L591
https://github.com/microsoft/fluentui/blob/ac24a6a79a7f09559bb35ffbc5b28f53ecd9e95f/packages/react-focus/src/components/FocusZone/FocusZone.tsx#L475

The others are just passed into the HTML element:
https://github.com/microsoft/fluentui/blob/ac24a6a79a7f09559bb35ffbc5b28f53ecd9e95f/packages/react-focus/src/components/FocusZone/FocusZone.tsx#L262

So why include `| FocusZone` in the type?

**Comment by JustSlone** [comment]
@khmakoto may be able to help answer this

**Reply by khmakoto** [reply]
I'm not sure of the historical reasons for this but it seems like we could remove that type?

What do you think @dzearing?

**Reply by khmakoto** [reply]
So, after speaking offline with @dzearing and drilling into `@types/react` this seems safe to remove. I'll spin up a PR soon and update here when I do.

**Comment by khmakoto** [comment, marked as answer]
Ok, this is going to be fixed in #16952.

Let us know if there is anything else I can help you with.

**Reply by Sentri** [reply]
Awesome! Thank you.
